src/server/routes/entity/entity.js

```diff
@@ -274,6 +273,136 @@ export function addNoteToRevision(req: PassportRequest, res: $Response) {
+	const entityHeader = await orm.Entity.forge({bbid}).fetch({transacting});
+
+	const model = utils.getEntityModelByType(orm, entityHeader.get('type'));
+	return model.forge({bbid}).fetch({transacting});
+}
+
+async function setParentRevisions(transacting, newRevision, parentRevisionIDs) {
+	if (_.isEmpty(parentRevisionIDs)) {
+		return Promise.resolve(null);
+	}
+
+	// Get the parents of the new revision
+	const parents =
+		await newRevision.related('parents').fetch({transacting});
+
+	// Add the previous revision as a parent of this revision.
+	return parents.attach(parentRevisionIDs, {transacting});
+}
+
+async function saveEntitiesAndFinishRevision(
+	orm, transacting, isNew: boolean, newRevision: any, mainEntity: any,
+	updatedEntities: [], editorID: number, note: string
+) {
+	const parentRevisionIDs = _.compact(_.uniq(updatedEntities.map(
+		(entityModel) => entityModel.get('revisionId')
+	)));
+
+	const entitiesSavedPromise = Promise.all(
+		_.map(updatedEntities, (entityModel) => {
+			entityModel.set('revisionId', newRevision.get('id'));
+
+			const shouldInsert =
+				entityModel.get('bbid') === mainEntity.get('bbid') && isNew;
+			const method = shouldInsert ? 'insert' : 'update';
+			return entityModel.save(null, {method, transacting});
+		})
+	);
+
+	const editorUpdatePromise =
+		utils.incrementEditorEditCountById(orm, editorID, transacting);
+
+	const notePromise = _createNote(
+		orm, note, editorID, newRevision, transacting
+	);
+
+	const parentsAddedPromise =
+		setParentRevisions(transacting, newRevision, parentRevisionIDs);
+
+	/** model.save returns a refreshed model */
+	const [savedEntities, ...others] = await Promise.all([
+		entitiesSavedPromise,
+		editorUpdatePromise,
+		parentsAddedPromise,
+		notePromise
+	]);
+
+	return savedEntities.find(entityModel => entityModel.get('bbid') === mainEntity.get('bbid')) || mainEntity;
+}
+
+async function deleteRelationships(orm, transacting, mainEntity) {
+	const mainBBID = mainEntity.bbid;
+	const {relationshipSet} = mainEntity;
+	const otherBBIDs = [];
+	const otherEntities = [];
+
+	if (relationshipSet) {
+		// Create a list of BBID's that is related to the deleted entity
+		relationshipSet.relationships.forEach((relationship) => {
+			if (relationship.sourceBbid === mainBBID) {
+				otherBBIDs.push(relationship.targetBbid);
+			}
+			else if (relationship.targetBbid === mainBBID) {
+				otherBBIDs.push(relationship.sourceBbid);
+			}
+		});
+
+		// Loop over the BBID's of other entites related to deleted entity
+		if (otherBBIDs.length) {
+			if (otherBBIDs.length === 0) {
+				return [];
+			}
+			await Promise.all(otherBBIDs.map(async (entityBbid) => {
+				const otherEntity = await getEntityByBBID(orm, transacting, entityBbid);
+
+				const otherEntityRelationshipSet = await otherEntity.relationshipSet()
+					.fetch({require: false, transacting, withRelated: 'relationships'});
+
+				if (_.isNull(otherEntityRelationshipSet)) {
```

Comment:
`_.isNil` allows to check in one go if something is null OR undefined.
I usually use that instead of `isNull` because javascript is pretty loose when it comes to what it returns.
Although in the cases I've tried isNull seemed to work, Id suggest erring on the safer side, since an `undefined` otherEntityRelationshipSet would also throw an error below.